include/particles/particle.tcc

```diff
@@ -224,20 +230,29 @@ mpm::HDF5Particle mpm::Particle<Tdim>::hdf5() const {
 // Initialise particle properties
 template <unsigned Tdim>
 void mpm::Particle<Tdim>::initialise() {
-  displacement_.setZero();
   dstrain_.setZero();
-  mass_ = 0.;
   natural_size_.setZero();
   set_traction_ = false;
   size_.setZero();
   strain_rate_.setZero();
   strain_.setZero();
   stress_.setZero();
   traction_.setZero();
-  velocity_.setZero();
-  volume_ = std::numeric_limits<double>::max();
   volumetric_strain_centroid_ = 0.;
 
+  // Initialize scalar properties
+  scalar_properties_.resize(3);
+  scalar_properties_.at(mpm::properties::Scalar::Mass) = 0.;
+  scalar_properties_.at(mpm::properties::Scalar::MassDensity) = 0.;
+  scalar_properties_.at(mpm::properties::Scalar::Volume) =
+      std::numeric_limits<double>::max();
```

Comment:
@kks32 I think this could be a problem. Currently, in `mpm_properties`, `Mass`, `Volume`, and `MassDensity` are defined with index 0, 1, and 2, which is safe for this case. But this will be an error if we want to add another variable which enum is larger than the size that you specified in `scalar_properties_.resize(size);`. It will be a memory access error.